### DISCUSSION TITLE
Get Vault Data in .net application

### DISCUSSION THREAD

**Discussion by mkeuschn** [dani-garcia/vaultwarden#4544] [Q&A] [answered]
I want to get vault data from my .net application. I have found a BitWarden Sdk (https://github.com/bitwarden/sdk/tree/main/languages/csharp#readme). But if I want to use this lib I need an access token from a machine account (https://bitwarden.com/help/machine-accounts/). I can't find the machine account option in Vault Warden. Is there another possibility to get the data? Or is it possible to create an access token?

**Comment by stefan0xC** [comment, marked as answer]
Vaultwarden does not implement the [Secrets Manager](https://bitwarden.com/products/secrets-manager/) API ([much of which is released](https://github.com/bitwarden/server/tree/main/bitwarden_license/src/Commercial.Core/SecretsManager) [only under a source-available license](https://github.com/bitwarden/clients/tree/main/bitwarden_license/bit-web/src/app/secrets-manager)) so the SDK is not the right tool.

For another possibility you might want to check out the [Vault Management API](https://bitwarden.com/help/vault-management-api/).
